packages/package-json-lint/src/rules/require-no-unnecessary-dependency.js

```diff
@@ -0,0 +1,131 @@
+const devDependencySet = [
```

Comment:
I have taken this set from devDependencies in packages in terra-toolkit and terra-core. 
Please let me know if there are any packages that need to be added or removed.
Thanks.

Reply:
can we have a separate file for this ? as the list might grow.

Reply:
I don't think that is required as later many rules may have large sets. And having a different file for each of them and maintaining them seems unnecessary.

Let me know if you think otherwise?